scipy/optimize/tests/test__numdiff.py

```diff
@@ -333,5 +359,104 @@ def test_check_derivative(self):
         assert_(accuracy == 0)
 
 
+class TestApproxDerivativeSparse(object):
+    # Example from Numerical Optimization
+    def __init__(self):
```

Comment:
That's a thick book :-). More specific reference (chapter? Example number, if any) would be helpful.